### PR TITLE
Advanced imputation metrics

### PR DESCRIPTION
**Pull request by lynnjo**
## Description

Added HapidSampleTable and SampleHapidByRange commands to aid with imputation analysis.

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [x] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Added hapid-sample-table and sample-hapid-by-range commands to aid with imputation analysis.
```

### CODE REVIEW

File: src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+            val outputFile = "$outputDir/export_gvcf_${region.replace(":","_")}.vcf"
+            var dataCommand = if (samples.getExportCommand()[0] == SampleFormatEnum.FILE.toString()) mutableListOf(
+                "tiledbvcf",
+                "export",
+                "--uri",
+                "$dbPath/gvcf_dataset",
+                "-m",
+                "-b",
+                "65536",
```
[the remark this answers is not in the document]
Comment (btmonier) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt:91]: @smm477 - [memory budget](https://docs.tiledb.com/main/integrations-and-extensions/genomics/population-genomics/api-reference/cli#options-2).

[the remark this answers is not in the document]
Comment (btmonier) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt:91]: I'm not sure what `-m` does though. It doesn't appear to be a exported flag according to the docs.

File: src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                "-O",
+                "v",
+                "--sample-names",
+                samples.getExportCommand()[1],
+                "-o",
+                outputFile,
+                "--regions",
+                region,
+                "--tiledb-config",
+                "sm.memory_budget=10737418240,sm.memory_budget_var=21474836480,sm.skip_unary_partitioning_budget_check=true"
```
Comment (btmonier) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt:119]: Maybe add some comments about the rationale behind these values?

**Review by smm477 — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt (hunk 2 above)
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt:119]: yes - I will add comments related to the tiledb-config options

File: src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt:91]: -m merges the samples to a single vcf.  Without this, you get a single file per sample.  IF you run help on tiledbvcf export you can see it listed.

[the remark this answers is not in the document]
Comment (btmonier) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt:91]: Well, looks like someone needs to update their CLI API docs!

**Review by tcasstevens — APPROVED** [review]

**Pull request merged**